src/main/java/spireMapOverhaul/zones/characterinfluence/CharacterInfluenceZone.java

```diff
@@ -0,0 +1,150 @@
+        AbstractZone currentZone = ZonePatches.currentZone();
+
+        if (!(currentZone instanceof CharacterInfluenceZone) ||
+                ((CharacterInfluenceZone) currentZone).classInfluence == null)
+            return null;
+
+        return ((CharacterInfluenceZone) currentZone).classInfluence;
+    }
+
+    @Override
+    public void mapGenDone(ArrayList<ArrayList<MapRoomNode>> map) {
+        super.mapGenDone(map);
+        do {
+            this.classInfluence = CardCrawlGame.characterManager.getRandomCharacter(AbstractDungeon.mapRng);
+        } while (this.classInfluence.chosenClass == AbstractDungeon.player.chosenClass && CardCrawlGame.characterManager.getAllCharacters().size() != 1);
+        this.name = TEXT[2] + this.classInfluence.title;
+        updateDescription();
+    }
+
+    @Override
+    public AbstractZone copy() {
+        return new CharacterInfluenceZone(this.classInfluence);
+    }
+
+    @Override
+    public Color getColor() {
+        if (this.classInfluence == null) return Color.GRAY;
+        return this.classInfluence.getCardRenderColor();
+    }
+
+    public void initPools() {
+        commonPool = new CardGroup(CardGroup.CardGroupType.CARD_POOL);
+        uncommonPool = new CardGroup(CardGroup.CardGroupType.CARD_POOL);
+        rarePool = new CardGroup(CardGroup.CardGroupType.CARD_POOL);
+
+        ArrayList<AbstractCard> tmpCardList = new ArrayList<>();
+        this.classInfluence.getCardPool(tmpCardList);
+
+        for (AbstractCard c : tmpCardList) {
+            switch (c.rarity) {
+                case COMMON:
+                    commonPool.addToTop(c);
+                    continue;
+                case UNCOMMON:
+                    uncommonPool.addToTop(c);
+                    continue;
+                case RARE:
+                    rarePool.addToTop(c);
+                    continue;
+                case CURSE:
+            }
+        }
+
+        //Now, we should have gotten our pools.
+
+    }
+
+    public AbstractCard getCard(AbstractCard.CardRarity rarity, AbstractCard.CardType type) {
+        if (commonPool == null || uncommonPool == null || rarePool == null) initPools();
+        switch (rarity) {
+            case UNCOMMON:
+                return type != null ? uncommonPool.getRandomCard(type, true) : uncommonPool.getRandomCard(true);
+            case RARE:
+                return type != null ? rarePool.getRandomCard(type, true) : rarePool.getRandomCard(true);
+            case COMMON:
+            default:
+                return type != null ? commonPool.getRandomCard(type, true) : commonPool.getRandomCard(true);
+        }
+    }
+
+    public void replaceCards(ArrayList<AbstractCard> cardList) {
+        ArrayList<AbstractCard> newCards = new ArrayList<>();
+        for (AbstractCard card : cardList) {
+            boolean dupe;
+            do {
+                dupe = false;
+                AbstractCard cardToAdd = getCard(card.rarity, card.type);
```

Comment:
Some characters have new card types, those would cause null to be returned here because tmp of CardGroup:getRandomCard would be empty.

Maybe account for that in your getCardMethod